transmart-core-db/grails-app/services/org/transmartproject/db/pedigree/RelationService.groovy

```diff
@@ -0,0 +1,27 @@
+package org.transmartproject.db.pedigree
+
+import grails.transaction.Transactional
+import groovy.transform.CompileStatic
+import org.springframework.beans.factory.annotation.Autowired
+import org.transmartproject.core.exceptions.AccessDeniedException
+import org.transmartproject.core.pedigree.RelationResource
+import org.transmartproject.core.users.User
+import org.transmartproject.db.ontology.MDStudiesService
+
+@Transactional(readOnly = true)
+@CompileStatic
+class RelationService implements RelationResource {
+
+    @Autowired
+    MDStudiesService studiesService
+
+    @Override
+    List<org.transmartproject.core.pedigree.Relation> getAll(User user) throws AccessDeniedException {
```

Comment:
In TranSMART things are typically done like this: service methods return domain object (Hibernate wrapper objects), but referred to by some interface. But still all the session stuff and lazy evaluation magic is still there.
What we do in other project (e.g., whereabouts) could be applied here to: stream to RelationRepresentation objects here (preferably using Java stream() instead of groovy (dynamic) collect).
I know that we deviate from other places in TranSMART if we start doing that, but I do think it is a change in the right direction.